### DISCUSSION TITLE
Include mainstream social icons?

### DISCUSSION THREAD

**Discussion by leekelleher** [umbraco/Umbraco-CMS#12650] [Features and ideas]
Would we consider including some of the mainstream social network icons with Umbraco? e.g. Facebook, Twitter, LinkedIn, et al.

Reasons for this...

1. Since we have been able to add custom SVG icons in the backoffice, for the majority of my Umbraco implementations I have included a set of mainstream social icons. These were used for various social content, footer links, etc, (mostly using the [Icon Picker editor from my Contentment package](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/icon-picker.md)). I'm sure others are doing similar.

2. When using the external login providers feature, the icons used are from the ["bootstrap-social" CSS dependency](https://github.com/umbraco/Umbraco-CMS/blob/release-10.0.1/src/Umbraco.Web.UI.Client/gulp/tasks/dependencies.js#L149), _(which itself depends on the FontAwesome library)._ With the ongoing push towards using SVG icons over font-based icons, this could be another step forwards.

As for which icons to include?

My initial suggestion would be, _(please excuse my western/UK bias)..._ [Facebook](https://fontawesome.com/icons/facebook?s=brands), [GitHub](https://fontawesome.com/icons/github?s=brands), [Google](https://fontawesome.com/icons/google?s=brands), [Instagram](https://fontawesome.com/icons/instagram?s=brands), [LinkedIn](https://fontawesome.com/icons/linkedin?s=brands), [Twitter](https://fontawesome.com/icons/twitter?s=brands), [YouTube](https://fontawesome.com/icons/youtube?s=brands)
...but we are an international audience, what else could be considered mainstream?

I'm not suggesting include _all_ of FontAwesome's SVG brands library, only a subset of the mainstream platforms.
In a similar vein to the [oEmbed providers](https://github.com/umbraco/Umbraco-CMS/blob/release-10.0.1/src/Umbraco.Core/DependencyInjection/UmbracoBuilder.Collections.cs#L109-L123), a curated awareness of external web services/platforms.

**Comment by FransdeJong** [comment]
Great suggestion. I think a lot of us include them anyway so why not include them in the default set.

**Comment by markadrake** [comment]
One of the first tasks we do in each Umbraco solution is to add the icon library from [SimpleIcons](https://github.com/simple-icons/simple-icons).
